filter_plugins/linux_ufw_expand_rules.py

```diff
@@ -0,0 +1,237 @@
+import unittest
+from copy import deepcopy
+from typing import List, Dict
+
+
+class LinuxUFWExpandRules:
+    @staticmethod
+    def expand_rules(rules: List) -> List:
+        rules = LinuxUFWExpandRules._expand_rules(rules, 'from_ip', ['from'])
+        rules = LinuxUFWExpandRules._expand_rules(rules, 'from_port')
+        rules = LinuxUFWExpandRules._expand_rules(rules, 'to_ip', ['to'])
+        rules = LinuxUFWExpandRules._expand_rules(rules, 'to_port', ['port'])
+        rules = LinuxUFWExpandRules._expand_rules(rules, 'interface')
+        rules = LinuxUFWExpandRules._expand_rules(rules, 'proto')
+        return rules
+
+    @staticmethod
+    def _expand_rules(rules: List, key: str, key_aliases=[]) -> List:
```

Comment:
private, please "__" instead of "_"

Reply:
The method is for internal use, so `_` is correct (see PEP 8, the section about naming styles). `__` (without trailing `__`) would enable name mangling, which is strongly discouraged by many and generally not considered _pythonic_. (Also name mangling does not make much sense if the method is `@staticmethod`.)

Reply:
"There is some controversy about the use of __names.

If your class is intended to be subclassed, and you have attributes that you do not want subclasses to use, consider naming them with double leading underscores and no trailing underscores.

Note that only the simple class name is used in the mangled name, so if a subclass chooses both the same class name and attribute name, you can still get name collisions.

Name mangling can make certain uses, such as debugging and __getattr__() , less convenient. However the name mangling algorithm is well documented and easy to perform manually.

Not everyone likes name mangling. Try to balance the need to avoid accidental name clashes with potential use by advanced callers."

I was referring to that dunder is considered as what is "private" in other languages, while a single underscore is consodered what is "protected" in other languages. I still use the notions of private and protected. And I don't propagate to the outside or subclasses what does not need to be propagated. Yes this prvents me from having class A subclassing Class A, but I take this inconvinience. Yes in this case it is most probably not that relevant.